### PR TITLE
Docs docs docs :)

### LINKED ISSUE #140: Correct the config sample
There are my personal path in it, it seemed obvious that they needed to be replaced but it's actually confusing to people.

### LINKED ISSUE #149: doc/demo.md with a comprehensive from scratch guide to set up a demo revault network

### PR DESCRIPTION
**Pull request by danielabrozzoni**
This PR needs https://github.com/revault/cosignerd/pull/12 merged and probably something in coordinator too :)

Closes #149
Closes #140

### CODE REVIEW

File: doc/USAGE.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+- Update the `data_dir`: this is the directory where all the data will be saved. We'll use `revaultd_data`
+```
+mkdir revaultd_data
+```
+- Update the `coordinator_host`, `coordinator_noise_key` with the data obtained in step 2
+- Update the `stakeholders_xpubs`: it will contain all the stakeholder xpubs created in the ceremony
+- Update the `managers_xpubs`: it will contain all the manager xpubs created in the ceremony
+- Update the `cosigner_keys`: it will contain all the cosigner keys created in step 3
+- Update the `stakeholder_config`: update `xpub` to match your xpub; this section must be there only if you're a stakeholder, or a manager-stakeholder. Leave the watchtowers section as it is, it's not used (yet!)
+- Update the `manager_config`: update `xpub` to match your xpub; the `cosigner` key to match your cosigner key otained in step 3; this section must be there only if you're a manager, or a manager-stakeholder
```
Comment (darosior) [inline doc/USAGE.md:106]: Maybe we should explicit they need to make a choice at the beginning ?

**Review by darosior — COMMENTED** [review]
Awesome. Thanks. That's exactly what i had in mind. Left a couple of comments, i think we can even get further in the explanations, @kloaec could help us to get what is not clear at first sight.

File: doc/USAGE.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
+# Tutorial: using revaultd for the first time!
+
+## Prerequisites
+- Bitcoin Core version >= 0.21.0
+- The Revault ceremony
+    - Revaultd supposes that you've already done a proper ceremony with all the stakeholders and managers, where you created the keys in a secure way and exchanged them.
+    - If you're just exploring revault you probably don't have time to do a proper ceremony: let's recreate something similar.
+    - First of all, it's important to clarify how many stakeholders and managers you'll have: you must have at least 2 stakeholders and 1 manager, and each partecipant gets its own computer. If you're by yourself but you still want to try out Revault, you can get around this by running multiple revaultd instances with multiple configuration files.
```
[the remark this answers is not in the document]
Comment (danielabrozzoni) [reply doc/USAGE.md:8]: Mamma mia 🤦🏻 🍕

File: doc/USAGE.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+cd revaultd
+```
+`revaultd` needs a configuration file - you can find [here](../contrib/config_regtest.toml) an example of configuration.
+The default path for the configuration is `~/.revault/revault.toml`, but you can put it whenever you want: we'll just pass a special flag to `revaultd` to instruct it on where to look. For this example, we'll copy the config into `our_modified_config.toml`:
+```
+cp contrib/config_regtest.toml our_modified_config.toml
+```
+We'll need to modify the config a bit to fit our example:
+- Set the `daemon` to `false`. We want to see if something crashes, at least for the first time!
+- Update the `data_dir`: this is the directory where all the data will be saved. We'll use `revaultd_data`
```
[the remark this answers is not in the document]
Comment (danielabrozzoni) [reply doc/USAGE.md:97]: Now that I think about it, I'm not sure you can use relative paths in the conf...

File: doc/USAGE.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+```
+cd revaultd
+```
+`revaultd` needs a configuration file - you can find [here](../contrib/config_regtest.toml) an example of configuration.
+The default path for the configuration is `~/.revault/revault.toml`, but you can put it whenever you want: we'll just pass a special flag to `revaultd` to instruct it on where to look. For this example, we'll copy the config into `our_modified_config.toml`:
+```
+cp contrib/config_regtest.toml our_modified_config.toml
+```
+We'll need to modify the config a bit to fit our example:
+- Set the `daemon` to `false`. We want to see if something crashes, at least for the first time!
```
[the remark this answers is not in the document]
Comment (danielabrozzoni) [reply doc/USAGE.md:107]: Fuck, rethinking about this. If I set it to `false` then I have to open another terminal and set the aliases again. 

Sorry Ed 💔

File: doc/USAGE.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+```
+
+### 2. Starting the coordinator
+Cd into the coordinator:
+```
+cd coordinatord
+```
+TBD: docker script?
+
+After starting the coordinator, please keep note of the host and the noise key used, we'll need them later.
```
Comment (darosior) [inline doc/USAGE.md:55]: You should tell them to stop it imo

File: doc/USAGE.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+├── cosignerd
+├── revaultd
+├── revault-gui
+```
+
+## Let's get started!
+### 1. Spinning up Bitcoin Core
+In this tutorial we'll use the regtest network. We'll use a custom data directory inside the `revault_tutorial`.
+```
+mkdir revault_core_data
```
Comment (darosior) [inline doc/USAGE.md:40]: nit: may i suggest `bitcoind_datadir` ? It seems more explicit to me

File: doc/USAGE.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+```
+cd cosignerd
+```
+
+You can find an [example config](https://github.com/revault/cosignerd/tree/master/contrib/config_regtest.toml) to begin with. Copy it to `./our_modified_config.toml`
+```
+cp contrib/config_regtest.toml our_modified_config.toml
+```
+We'll need to modify it a bit:
+- Update the manager xpubs with the xpubs generated at the ceremony.
```
Comment (darosior) [inline doc/USAGE.md:73]: No need for the managers xpubs on the cosigning servers

File: doc/USAGE.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+To make things easier, we'll create a couple of aliases:
+```
+alias rd="cargo run --release --bin revaultd -- --conf our_modified_config.toml"
+alias rcli="cargo run --release --bin revault-cli -- --conf our_modified_config.toml"
+```
+Now starting `revaultd`:
+```
+rd
+```
+Revaultd will print the noise key used when starting. Collect all the managers' noise keys, put them in the cosigner configuration (the one we created in step 2).
```
Comment (darosior) [inline doc/USAGE.md:131]: Maybe make "cosigner" plural to emphase they need to copy it everywhere?

File: doc/USAGE.md [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+
+If no error was printed, congrats! You succesfully started `revaultd`.
+
+Now kill it using `CTRL+C`, open the `revaultd` configuration again and set `daemon=true`: we don't want to occupy and entire terminal for `revaultd`!
```
Comment (darosior) [inline doc/USAGE.md:135]: Haha, no we don't!

File: doc/USAGE.md [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+Let's go back to the revaultd directory and ask for info:
+```
+cd revaultd
+rcli getinfo
+```
+
+You should see something like:
+```
+{"id":"revault-cli-7957","jsonrpc":"2.0","result":{"blockheight":172,"network":"regtest","sync":1.0,"vaults":1,"version":"0.0.2"}}
+```
```
Comment (darosior) [inline doc/USAGE.md:170]: We can update it now that #44 is merged :)
```
{
  "id": "revault-cli-150897",
  "jsonrpc": "2.0",
  "result": {
    "blockheight": 220,
    "network": "regtest",
    "sync": 1.0,
    "vaults": 5,
    "version": "0.0.2"
  }
}
```

File: doc/USAGE.md [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+
+### 7. Wait, CLI only?!
+Nah. We also have a GUI. Check out [revault-gui](https://github.com/revault/revault-gui) :)
```
Comment (darosior) [inline doc/USAGE.md:173]: Great

File: doc/USAGE.md [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+
+## Testing
+For testing you may find useful the [regtest manager script](../contrib/regtest_manager.sh). You will need to tweak the configuration we created in step 4 a bit, as the RPC address of the bitcoin node spawn by the script is `127.0.0.1:9001`.
```
Comment (darosior) [inline doc/USAGE.md:176]: I wanted to get rid of this script in the repo as it seems to only be useful to me and confusing to others, but finally maybe not?

**Review by darosior — COMMENTED** [review]
Some more comments. This is really looking good

File: doc/USAGE.md (hunk 12 above)
Comment (danielabrozzoni) [reply doc/USAGE.md:176]: Not sure if it's actually useful to people (I've never tried it <3), but it's not hurting anyone, so I would just keep it there

File: README.md [hunk 13]
```diff
@@ -18,20 +18,7 @@ the coins stored and may or may not own them) logic is part of this daemon.
...
-$ re stop
-```
```
[the remark this answers is not in the document]
Comment (danielabrozzoni) [reply README.md:34]: It's a move not a deletion! I swear! 😄

File: doc/USAGE.md (hunk 7 above)
Comment (danielabrozzoni) [reply doc/USAGE.md:73]: LOL

I really think that we should display a warning when there's stupid shit in the configuration, I've kept them there for weeks

File: doc/USAGE.md [hunk 14]
```diff
@@ -0,0 +1,187 @@
+# Tutorial: using revaultd for the first time!
+
+## Prerequisites
+- Bitcoin Core version >= 0.21.0
```
Comment (danielabrozzoni) [inline doc/USAGE.md:4]: TODO for myself: Use headers instead of bullet points, this is not Roam

File: doc/USAGE.md [hunk 15]
```diff
@@ -0,0 +1,187 @@
...
+cd cosignerd_data
+python3 -c 'import os;open("bitcoin_secret", "wb").write(bytes(1) + os.urandom(31))'
+cd ..
+```
+
+Now start the cosignerd:
+```
+cargo run --release -- --conf our_modified_config.toml
+```
+It will print a Bitcoin public key. Save it, we'll need it later, then kill the cosignerd with `CTRL-C`.
```
[the remark this answers is not in the document]
Comment (kloaec) [reply doc/USAGE.md:90]: Please add a comment on the output, right now the key is just displayed, without saying it's the key.

**Review by kloaec — COMMENTED** [review]
Partial review, i'll get back at it later :)

File: doc/USAGE.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (darosior) [reply doc/USAGE.md:90]: Done in https://github.com/revault/cosignerd/pull/13
New output:
```
[2021-04-09][10:35:51][cosignerd][INFO] Started cosignerd daemon with Noise pubkey 'a5e74bfb27cb715e7bbedca00f85d1b09b05eeab96acb7cec26bfc4ef18c923c' and Bitcoin pubkey '02ba13edd7fc401641f0aa5fc504ff5b55ddd92954e873f7c1c3cbd02c771300b4'
```

File: README.md [hunk 16]
```diff
@@ -18,20 +18,7 @@ the coins stored and may or may not own them) logic is part of this daemon.
...
+Please see [`doc/USAGE.md`](doc/USAGE.md) if you want a tutorial on how to deploy revaultd.
```
Comment (danielabrozzoni) [inline README.md:21]: TODO: either here or in `USAGE.md`, write a bit about the toolchain that must be used (I remember I had some troubles when using rust 1.46 I think)

**Comment by danielabrozzoni** [comment]
Still needs the coordinatord docker script merged

**Comment by darosior** [comment]
Ok, apart from that is it in a reviewable state ? Can still give it a look in the meanwhile

**Comment by danielabrozzoni** [comment]
Yes it is :)

File: doc/USAGE.md [hunk 17]
```diff
@@ -0,0 +1,572 @@
...
+See https://bitcoin.org/en/download
+
+`revaultd` **won't work** with previous versions! Check your version with `bitcoin-cli --version`
+
+### Rust
+`revaultd`, `cosignerd`, `coordinatord` and `revault-gui` are written in Rust. Since we don't publish compiled binaries (yet!), you'll need to have the Rust toolchain installed to be able to compile the projects.
+
+Please refer to https://www.rust-lang.org/tools/install for instructions on how to install Rust.
+
+Double check that your rust toolchain is **at least** 1.46.0:
```
Comment (danielabrozzoni) [inline doc/USAGE.md:29]: FIXME: actually it should be 1.49 for the GUI

**Comment by darosior** [comment]
https://github.com/revault/coordinatord/pull/10 is now merged

File: doc/USAGE.md [hunk 18]
```diff
@@ -0,0 +1,572 @@
...
+```
+mkdir revault_tutorial
+cd revault_tutorial
+```
+Download all the needed repositories: `cosignerd`, `coordinatord`, `revaultd` and optionally, `revault-gui`:
+```
+git clone https://github.com/revault/coordinatord
+git clone https://github.com/revault/cosignerd
+git clone https://github.com/revault/revaultd
+git clone https://github.com/revault/revault-gui
```
Comment (darosior) [inline doc/USAGE.md:72]: But `revault-gui` is outside of the scope here?

File: doc/USAGE.md [hunk 19]
```diff
@@ -0,0 +1,572 @@
...
+### 3. Configuring the cosigner
+We'll need one cosigner for each stakeholder.
+Cd into the cosignerd:
+```
+cd cosignerd
+```
+
+Let's start by compiling the project (it may take a while):
+```
+cargo build --release
```
Comment (darosior) [inline doc/USAGE.md:131]: nit: no need for `--release` for the cosigner

File: doc/USAGE.md [hunk 20]
```diff
@@ -0,0 +1,167 @@
...
+cd cosignerd
+```
+
+You can find an [example config](https://github.com/revault/cosignerd/tree/master/contrib/config_regtest.toml) to begin with. Copy it to `our_modified_config.toml`
+```
+cp contrib/config_regtest.toml our_modified_config.toml
+```
+We'll need to modify it a bit:
+- Update the manager xpubs with the xpubs generated at the ceremony.
+- Don't bother with the noise keys now, we'll have to update them later.
```
[the remark this answers is not in the document]
Comment (darosior) [reply doc/USAGE.md:146]: ^

File: doc/USAGE.md [hunk 21]
```diff
@@ -0,0 +1,572 @@
...
+cargo build --release
+```
+
+#### Stakeholder 1
+First of all, create a directory to store all the cosignerd data:
+```
+mkdir cosignerd_1_data
+```
+
+You can find an [example config](https://github.com/revault/cosignerd/tree/master/contrib/config_regtest.toml) to begin with. Copy it to `./cosigner_1_config.toml`
```
Comment (darosior) [inline doc/USAGE.md:140]: I'm going to tag a `demo`

File: doc/USAGE.md (hunk 17 above)
Comment (darosior) [reply doc/USAGE.md:29]: And it's `1.43` for `revaultd` :p

File: doc/USAGE.md [hunk 22]
```diff
@@ -0,0 +1,572 @@
...
+```
+### 4. Creating the revaultd configuration
+Cd into the project:
+```
+cd revaultd
+```
+
+Build it:
+```
+cargo build --release
```
Comment (darosior) [inline doc/USAGE.md:215]: nit: no need for `--release`

File: doc/USAGE.md [hunk 23]
```diff
@@ -0,0 +1,572 @@
...
+```
+cd cosignerd
+```
+
+Update the configuration we created in step 2 (`cosigner_1_config` and `cosigner_2_config`) with the manager noise key we just obtained.
+
+Now it's time to start the cosigners with the right config:
+```
+cargo run --release -- --conf cosigner_1_config.toml
+cargo run --release -- --conf cosigner_2_config.toml
```
Comment (darosior) [inline doc/USAGE.md:461]: nit: no need for `--release`

File: doc/USAGE.md [hunk 24]
```diff
@@ -0,0 +1,572 @@
...
+stake_1_cli getinfo
+```
+
+You should see something like:
+```
+{
+    "blockheight": 220,
+    "network": "regtest",
+    "sync": 1.0,
+    "vaults": 5,
```
Comment (darosior) [inline doc/USAGE.md:500]: `0` ?

File: doc/USAGE.md [hunk 25]
```diff
@@ -0,0 +1,572 @@
...
+```
+
+and then
+
+```
+man_1_cli getdepositaddress
+```
+
+If they all return the same address, cool, somehow you managed to not fuck it up. Otherwise, well, you fucked up.
+Don't worrry, it happens. We're working towards making it happen less often, but in the meanwhile, you'll have to debug.
```
Comment (darosior) [inline doc/USAGE.md:528]: Nice, please disregard my earlier comment and mention descriptors there?

File: doc/USAGE.md [hunk 26]
```diff
@@ -0,0 +1,572 @@
...
+Restart `revaultd` and check again the deposit addresses. If that didn't work either, try restarting `bitcoind`.
+
+### 10. Wait, CLI only?!
+Nah. We also have a GUI. Check out [revault-gui](https://github.com/revault/revault-gui) :)
+
+## Organizing the perfect Revault party
+
+Want to try out Revault on multiple computers? You can follow this tutorial, but please be mindful that:
+1. You'll need to be on the same Bitcoin network - either one common regtest/signet, or testnet.
+2. you'll need to reach the `cosignerd` and the `coordinatord` from outside `localhost`. You'll have to update the configurations to make sure that the listen address is #FIXME I have no idea. 0.0.0.0?
```
Comment (darosior) [inline doc/USAGE.md:557]: > You'll have to make sure to not bind on `localhost` (eg use `0.0.0.0:<port>` to listen on all interfaces.

**Review by darosior — COMMENTED** [review]
Looks good to me. A few comments. Will test as soon as it's final

File: doc/USAGE.md [hunk 27]
```diff
@@ -0,0 +1,572 @@
...
+Please refer to https://www.rust-lang.org/tools/install for instructions on how to install Rust.
+
+Double check that your rust toolchain is **at least** 1.46.0:
+```
+cargo --version
+```
+
+### Some text editor
+There are various keys that you'll need to keep during the whole configuration. Obviously a text editor is not a good choice for storing real keys, but hey, we're in regtest. Open your favorite note-taking app (even Microsoft Word is fine, or even worse, Emacs)
+
```
Comment (kloaec) [inline doc/USAGE.md:36]: /!\ this is the first time we mention this is a test environment. IMHO it should be at the very top of the document OR if it's not about the demo, then the Regtest mention should be removed.
TLDR: to me it's unclear if USAGE.md is about usage of Revaultd, or the Demo :)

File: README.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (kloaec) [reply README.md:21]: ACK, had the same comment.
Also: we should mention that the shell is for LINUX, not sure the commands work on Windows

File: doc/USAGE.md (hunk 27 above)
Comment (danielabrozzoni) [reply doc/USAGE.md:36]: Agree, do you think it makes sense to move it to `doc/demo.md`?

Comment (darosior) [reply doc/USAGE.md:36]: YES :tm: `doc/DEMO.md`

File: README.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (danielabrozzoni) [reply README.md]: Specified that it's linux in `doc/DEMO.md`, solving :)

File: doc/USAGE.md [hunk 28]
```diff
@@ -0,0 +1,572 @@
...
+Every stakeholder will need:
+- `revaultd`
+- `cosignerd`
+- `revault-gui` (optional)
+
+Every manager will need:
+- `revaultd`
+- `revault-gui` (optional)
+
+Also, there must be **one** `coordinatord`
```
[the remark this answers is not in the document]
Comment (danielabrozzoni) [reply doc/USAGE.md]: Sure!

File: doc/USAGE.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (danielabrozzoni) [reply doc/USAGE.md]: Tested, they work *happy noises*

File: doc/DEMO.md [hunk 29]
```diff
@@ -0,0 +1,629 @@
...
+
+```
+cosigner_keys = [
+    *first cosigner key*,
+    *second cosigner key*,
+]
+```
+Please make sure that the keys are in the right order, the first as first and the second as second. Not doing so could lead to tricky bugs.
+
+- Update the `stakeholder_config`: update `xpub` to match the second stakeholder's xpub; leave the watchtowers section as it is, it's not used (yet!)
```
Comment (darosior) [inline doc/DEMO.md:319]: ditto: They should also remove the `manager_config`

File: contrib/config_regtest.toml [hunk 30]
```diff
@@ -15,34 +21,57 @@ stakeholders_xpubs = [
...
+# This option MUST NOT be changed after running revaultd for the first time. If you want to change it, please remove the following files:
+# - /path/to/your/data_dir/revaultd.sqlite3
+# - /path/to/your/data_dir/revaultd-watchonly-wallet-1
 # xprv9yFvjFWhx6hEJ3srPPCMAhQyrUhwJ9joByaQcJJAC37HsavncgLzq5ppWeb6QmkYEaYDkiJf424zHzmcjXx3Fn6DZqKx9EKpiv94suYjZ6e
 xpub = "xpub6CFH8m3bnUFXWXxKVQjMXqMiQWYRhcTeZCW1QghmkNeGkPFwADfFNt9JMuW38MnYVSAV9eyqJ3A61kbsfC5PSCdkZWi7pD2L4jv6edaPxKp"
+# At the moment this is unused
 watchtowers = [ { host = "127.0.0.1:1", noise_key = "46084f8a7da40ef7ffc38efa5af8a33a742b90f920885d17c533bb2a0b680cb3" } ]
 emergency_address = "bcrt1qewc2348370pgw8kjz8gy09z8xyh0d9fxde6nzamd3txc9gkmjqmq8m4cdq"
 
+# This section must be copied only if you're a stakeholder. Put here your xpub and cosigner configuration
```
Comment (darosior) [inline contrib/config_regtest.toml:70]: Only if you are a manager *

File: doc/DEMO.md [hunk 31]
```diff
@@ -0,0 +1,629 @@
...
+```
+mkdir stake_2_data
+```
+
+`revaultd` needs a configuration file - you can find [here](../contrib/config_regtest.toml) an example of configuration.
+The default path for the configuration is `~/.revault/revault.toml`, but for this example we'll use `./stake_2_config.toml`:
+
+```
+cp contrib/config_regtest.toml stake_2_config.toml
+```
```
Comment (darosior) [inline doc/DEMO.md:286]: Trying this for real what i did is just `cp stake_1_config.toml stake_2_config.toml` and changing the stakeholder xpub

File: doc/DEMO.md [hunk 32]
```diff
@@ -0,0 +1,629 @@
...
+
+```
+cosigner_keys = [
+    *first cosigner key*,
+    *second cosigner key*,
+]
+```
+Please make sure that the keys are in the right order, the first as first and the second as second. Not doing so could lead to tricky bugs.
+
+- Update the `stakeholder_config`: update `xpub` to match the first stakeholder's xpub; leave the watchtowers section as it is, it's not used (yet!)
```
Comment (darosior) [inline doc/DEMO.md:271]: They should also remove the `manager_config`

File: doc/DEMO.md [hunk 33]
```diff
@@ -0,0 +1,629 @@
...
+We'll need to modify the config a bit to fit our example:
+- Update the `data_dir`: this is the directory where all the data will be saved. We'll use `./man_1_data`:
+- Update the `coordinator_host`, `coordinator_noise_key` with the data obtained in step 2
+- Update the `stakeholders_xpubs`: it will contain all the stakeholder xpubs created in the ceremony
+
+```
+stakeholder_xpubs = [
+    *first stakeholder xpub*,
+    *second stakeholder xpub*,
+]
```
Comment (darosior) [inline doc/DEMO.md:343]: Same here (it's tedious and error-prone to re-copy the whole config, let's just copy the previous one replacing `stakeholder_config` with `manager_config` ?

File: doc/DEMO.md [hunk 34]
```diff
@@ -0,0 +1,629 @@
...
+
+```
+cosigner_keys = [
+    *first cosigner key*,
+    *second cosigner key*,
+]
+```
+Please make sure that the keys are in the right order, the first as first and the second as second. Not doing so could lead to tricky bugs.
+
+- Update the `manager_config`: update `xpub` to match your first manager's xpub; the first `cosigner` key to match the first cosigner key otained in step 3, the first `cosigner` host to match `127.0.0.1:10001`, the second cosigner key to match the second cosigner key otained in step 3, the second `cosigner` host to match `127.0.0.1:10002`
```
Comment (darosior) [inline doc/DEMO.md:365]: They should also remove the `stakeholder_config`

File: doc/DEMO.md [hunk 35]
```diff
@@ -0,0 +1,629 @@
...
+### 5. Spinning up revaultd
+Time to start revaultd! This will create the database and the watchonly wallet. After you run `revaultd` for the first time, the database and the watchonly wallet must be deleted if you wish to change one or more of the managers/stakeholders/cosigners keys.
+To make things easier, we'll create a couple of aliases:
+```
+alias stake_1_d="cargo run --release --bin revaultd -- --conf stake_1_config.toml"
+alias stake_1_cli="cargo run --release --bin revault-cli -- --conf stake_1_config.toml"
+alias stake_2_d="cargo run --release --bin revaultd -- --conf stake_2_config.toml"
+alias stake_2_cli="cargo run --release --bin revault-cli -- --conf stake_2_config.toml"
+alias man_1_d="cargo run --release --bin revaultd -- --conf man_1_config.toml"
+alias man_1_cli="cargo run --release --bin revault-cli -- --conf man_1_config.toml"
```
Comment (darosior) [inline doc/DEMO.md:378]: nit: unnecessary `--release` here

File: doc/DEMO.md [hunk 36]
```diff
@@ -0,0 +1,629 @@
...
+Nah. We also have a GUI. Check out [revault-gui](https://github.com/revault/revault-gui) :)
+
+## Organizing the perfect Revault party
+
+Want to try out Revault on multiple computers? You can follow this tutorial, but please be mindful that:
+1. You'll need to be on the same Bitcoin network - either one common regtest/signet, or testnet.
+2. you'll need to reach the `cosignerd` and the `coordinatord` from outside `localhost`. You'll have to make sure to not bind on localhost (eg use `listen="0.0.0.0:<port>"` to listen on all interfaces)
+
+## Testing
+For testing you may find useful the [regtest manager script](../contrib/regtest_manager.sh). You will need to tweak the configuration we created in step 4 a bit, as the RPC address of the bitcoin node spawn by the script is `127.0.0.1:9001`.
```
Comment (darosior) [inline doc/DEMO.md:596]: And that you use the same listen port for the cosigners that i use in the script, too (10001)

Comment (danielabrozzoni) [reply doc/DEMO.md:586]: Ah damn! I'll use a different port for the cosigners then

**Comment by darosior** [comment]
I think you messed up the rebase on master :)

File: doc/DEMO.md [hunk 37]
```diff
@@ -0,0 +1,624 @@
...
+    "614ad96890c309b8da6915ddb9eb1135caf228120178833e70798f20e0783b16",
+]
+
+watchtowers = []
+```
+
+Now start the `coordinatord` again:
+
+```
+cargo run -- --config coordinatord_config.toml
```
Comment (darosior) [inline doc/DEMO.md:500]: You forgot `daemon = true`

File: doc/DEMO.md [hunk 38]
```diff
@@ -0,0 +1,624 @@
...
+```
+
+Go back to the parent directory for the next step
+```
+cd ../../
```
Comment (darosior) [inline doc/DEMO.md:505]: Just `cd ../`

File: doc/DEMO.md [hunk 39]
```diff
@@ -0,0 +1,624 @@
...
+```
+{
+  "result":{
+    "blockheight":215,
+    "network":"regtest",
+    "sync":1.0,
+    "vaults":2,
+    "version":"0.0.2"
+  }
+}
```
Comment (darosior) [inline doc/DEMO.md:533]:
```
{
  "result": {
    "blockheight": 0,
    "network": "regtest",
    "sync": 1.0,
    "vaults": 0,
    "version": "0.0.2"
  }
}
```

File: doc/DEMO.md [hunk 40]
```diff
@@ -0,0 +1,624 @@
...
+```
+rm ./stake_1_data/revaultd.sqlite3
+rm ./stake_2_data/revaultd.sqlite3
+rm ./man_1_data/revaultd.sqlite3
+rm -r ./stake_1_data/revaultd-watchonly-wallet-1/
+rm -r ./stake_2_data/revaultd-watchonly-wallet-1/
+rm -r ./man_1_data/revaultd-watchonly-wallet-1/
+```
+
+Restart `revaultd` and check again the deposit addresses. If that didn't work either, try restarting `bitcoind`.
```
Comment (darosior) [inline doc/DEMO.md:579]: > If that didn't work either, try restarting `bitcoind`.

If that didn't work either, open an issue or ask on `#revault` on freenode. :)

File: doc/DEMO.md [hunk 41]
```diff
@@ -0,0 +1,624 @@
...
+
+### 10. Wait, CLI only?!
```
Comment (darosior) [inline doc/DEMO.md:581]: Just before maybe point them to the list of available commands in `doc/API.md` ?

File: doc/DEMO.md [hunk 42]
```diff
@@ -0,0 +1,624 @@
...
+2. you'll need to reach the `cosignerd` and the `coordinatord` from outside `localhost`. You'll have to make sure to not bind on localhost (eg use `listen="0.0.0.0:<port>"` to listen on all interfaces)
+
+## Testing
+For testing you may find useful the [regtest manager script](../contrib/regtest_manager.sh). You will need to tweak the configuration we created in step 4 a bit, as the RPC address of the bitcoin node spawn by the script is `127.0.0.1:9001`.
+```
+$ . contrib/regtest_manager
+$ start_regtest
+$ fund_regtest
+$ start_revaultd
+$ re listvaults
```
Comment (darosior) [inline doc/DEMO.md:597]: I think you are missing `fund_vault` there

File: doc/DEMO.md [hunk 43]
```diff
@@ -0,0 +1,624 @@
...
+Nah. We also have a GUI. Check out [revault-gui](https://github.com/revault/revault-gui) :)
+
+## Organizing the perfect Revault party
+
+Want to try out Revault on multiple computers? You can follow this tutorial, but please be mindful that:
+1. You'll need to be on the same Bitcoin network - either one common regtest/signet, or testnet.
+2. you'll need to reach the `cosignerd` and the `coordinatord` from outside `localhost`. You'll have to make sure to not bind on localhost (eg use `listen="0.0.0.0:<port>"` to listen on all interfaces)
+
+## Testing
+For testing you may find useful the [regtest manager script](../contrib/regtest_manager.sh). You will need to tweak the configuration we created in step 4 a bit, as the RPC address of the bitcoin node spawn by the script is `127.0.0.1:9001`.
```
Comment (darosior) [inline doc/DEMO.md:591]: That's for testing without the network setup though

**Review by darosior — COMMENTED** [review]
Awesome. Could set it up by mostly blindly following the tutorial.

```
darosior@darosior:~/projects/revault_tutorial/revaultd$ stake_1_cli getdepositaddress
    Finished dev [unoptimized + debuginfo] target(s) in 0.03s
     Running `target/debug/revault-cli --conf stake_1_config.toml getdepositaddress`
{
  "result": {
    "address": "bcrt1qka86qt3lkns2qnxqyf5r4f93dek5zt9eyhsuhav5fvt5dvq6z2aqg6y7vc"
  }
}
darosior@darosior:~/projects/revault_tutorial/revaultd$ stake_2_cli getdepositaddress
    Finished dev [unoptimized + debuginfo] target(s) in 0.03s
     Running `target/debug/revault-cli --conf stake_2_config.toml getdepositaddress`
{
  "result": {
    "address": "bcrt1qka86qt3lkns2qnxqyf5r4f93dek5zt9eyhsuhav5fvt5dvq6z2aqg6y7vc"
  }
}
darosior@darosior:~/projects/revault_tutorial/revaultd$ man_1_cli getdepositaddress
    Finished dev [unoptimized + debuginfo] target(s) in 0.03s
     Running `target/debug/revault-cli --conf man_1_config.toml getdepositaddress`
{
  "result": {
    "address": "bcrt1qka86qt3lkns2qnxqyf5r4f93dek5zt9eyhsuhav5fvt5dvq6z2aqg6y7vc"
  }
}
```
:tada:

This is much better by copying configs :-)
A few comments still, but i think that's the last round!

**Comment by darosior** [comment]
One last thing: be sure to make them use the tags for each repo as `master`s are going to move fast

File: doc/DEMO.md [hunk 44]
```diff
@@ -0,0 +1,624 @@
...
+
+#### Stakeholder 1
+First of all, create a directory to store all the cosignerd data:
+```
+mkdir cosignerd_1_data
+```
+
+You can find an [example config](https://github.com/revault/cosignerd/tree/master/contrib/config_regtest.toml) to begin with. Copy it to `./cosigner_1_config.toml`
+```
+cp contrib/config_regtest.toml cosigner_1_config.toml
```
Comment (edouardparis) [inline doc/DEMO.md:171]: https://github.com/revault/cosignerd/tree/master/contrib
I see only a config.toml, not config_regtest.toml

File: doc/DEMO.md [hunk 45]
```diff
@@ -0,0 +1,624 @@
...
+We'll properly update the coordinator configuration later. For now, let's just retrieve the coordinator's noise key.
+Cd into the coordinator:
+```
+cd coordinatord
+```
+
+Build the project:
+```
+cargo build
+```
```
[the remark this answers is not in the document]
Comment (danielabrozzoni) [reply doc/DEMO.md:113]: I would prefer to keep them separate, so people who absolutely don't know how Rust works don't look for the noise keys through a bunch of "Downloading..." "Compiling..."

[the remark this answers is not in the document]
Comment (darosior) [reply doc/DEMO.md:113]: Good point

File: doc/DEMO.md [hunk 46]
```diff
@@ -0,0 +1,638 @@
...
+man_1_cli getdepositaddress
+```
+
+If they all return the same address, cool, somehow you managed to not fuck it up. Otherwise, well, you fucked up.
+Don't worrry, it happens. We're working towards making it happen less often by introducing descriptors in the configuration file, but in the meanwhile, you'll have to debug.
+
+If the deposit address are different, it means that the descriptors created by `revaultd` are different. Please check that:
+
+- The `stakeholder_xpubs` are all equal in all configurations, **the order of the keys MUST be the same**
+- The `manager_xpubs` are all equal in all configurations, **the order of the keys MUST be the same**
```
Comment (darosior) [inline doc/DEMO.md:571]: nit: missing an *s*

**Review by darosior — APPROVED** [review]
Impressive work. Don't have time to do another setup from scratch today but based on my Yesterday tests this looks pretty good to me.

ACK c773365

File: doc/DEMO.md [hunk 47]
```diff
@@ -0,0 +1,638 @@
...
+
+[[manager_config.cosigners]]
+host = "127.0.0.1:20001"
+noise_key = "2b1df6c0618cf54955046ca5ca1dc113ddc1d63e89074b3efefae5847b1d7a63"
+
+[[manager_config.cosigners]]
+host = "127.0.0.1:20002"
+noise_key = "f866b639cbd36fcf6c984bd70e1259aa4cad335c99a47ea3747d489f64d57e65"
+```
+- Update the `manager_config`: update `xpub` to match your first manager's xpub; the first cosigner key to match the first cosigner key otained in step 3, the second cosigner key to match the second cosigner key otained in step 3. Beware of the order of the keys! Swapping them could lead to nasty bugs.
```
Comment (edouardparis) [inline doc/DEMO.md:356]:
```suggestion
- Update the `manager_config`: update `xpub` to match your first manager's xpub; the first cosigner key to match the first cosigner key otained in step 3, the second cosigner key to match the second cosigner key obtained in step 3. Beware of the order of the keys! Swapping them could lead to nasty bugs.
```

**Comment by danielabrozzoni** [comment]
Solved nits

**Comment by edouardparis** [comment]
ACK 419382ebbb809364969c29372a1416f8e43d116b

**Comment by darosior** [comment]
ACK 419382e -- only change since last review are the typo fixes

**Comment by darosior** [comment]
What's with the CI again.... We should have a pytest timeout
